propolis/src/hw/nvme/cmds.rs

```diff
@@ -601,8 +607,27 @@ impl PrpIter<'_> {
                     // entry which should be present in PRP2
                     PrpNext::Prp2
                 } else {
-                    let list_off = (self.prp2 & PAGE_OFFSET as u64) / 8;
-                    PrpNext::List(self.prp2, list_off as u16)
+                    /*
+                     * The first PRP List entry (i.e. the first pointer to a memory page containing
+                     * additional PRP entries) that if present is contained in the PRP Entry 2 location
+                     * within the command, shall be Qword aligned and may also have a non-zero offset within
+                     * the memory page.
+                     */
+                    assert!((self.prp2 % 8) == 0);
```

Comment:
If this is a value controlled by the guest, then `assert`ing if it is not as-expected is rather dangerous, as broken behavior on their part leads to an immediate reset (propolis crash)

Reply:
Right, addressed in eb141578c3763d1e1ed035fe307159b20ae8e5fe